CDTDatastore.podspec

```diff
@@ -29,27 +29,53 @@ Pod::Spec.new do |s|
 
   s.ios.deployment_target = '6.0'
   s.osx.deployment_target = '10.8'
+
   s.requires_arc = true
 
-  s.source_files = 'Classes/**/*.{h,m}'
-  s.exclude_files = 'Classes/vendor/MYUtilities/*.{h,m}'
+  s.default_subspec = 'common'
+
+  s.subspec 'common' do |sp|
+    sp.prefix_header_contents = '#import "CollectionUtils.h"', '#import "Logging.h"', '#import "Test.h"'
+
+    sp.source_files = 'Classes/**/*.{h,m}'
+    sp.exclude_files = 'Classes/vendor/MYUtilities/*.{h,m}'
+
+    sp.ios.exclude_files = 'Classes/osx'
+    sp.osx.exclude_files = 'Classes/ios'
 
-  s.prefix_header_contents = '#import "CollectionUtils.h"', '#import "Logging.h"', '#import "Test.h"'
+    sp.frameworks = 'SystemConfiguration'
 
-  s.ios.exclude_files = 'Classes/osx'
-  s.osx.exclude_files = 'Classes/ios'
-  # s.public_header_files = 'Classes/common/CloudantSync.h'
+    sp.dependency 'CDTDatastore/no-arc'
+    sp.dependency 'CocoaLumberjack', '= 2.0.0-rc2'
 
-  s.dependency 'FMDB', '= 2.3'
-  s.dependency 'CocoaLumberjack', '= 2.0.0-rc2'
+    sp.default_subspec = 'SQLite'
 
-  s.frameworks = 'SystemConfiguration'
-  s.library = 'sqlite3', 'z'
+    sp.subspec 'SQLite' do |ssp|
+      ssp.library = 'sqlite3', 'z'
+      
+      ssp.dependency 'FMDB', '= 2.3'
+    end
+
+    sp.subspec 'SQLCipher' do |ssp|
+      ssp.library = 'z'
+
+      ssp.dependency 'FMDB/SQLCipher', '= 2.3'
+      ssp.xcconfig = { 'OTHER_CFLAGS' => '$(inherited) -DENCRYPT_DATABASE' }
+    end
```

Comment:
There is no easy solution for this problem.

Essentially `pod lib lint` checks that the subspecs compile, i.e:

- [SQLite](https://github.com/cloudant/CDTDatastore/pull/112/files#diff-06ec762bd04c5ee54e244d1e3fd0d111R53) depends only on **FMDB**, after downloading this source code, `pod lib lint` tries to compile it and if it succeeds, the validation passes.
- [common](https://github.com/cloudant/CDTDatastore/pull/112/files#diff-06ec762bd04c5ee54e244d1e3fd0d111R37) depends on [no-arc](https://github.com/cloudant/CDTDatastore/pull/112/files#diff-06ec762bd04c5ee54e244d1e3fd0d111R67) and **CocoaLumberjack** and the default subspec is `SQLite`. `pod lib lint` downloads and copies the necessary source code and then compiles; again if it succeeds, the validation passes.

Now, our CDT-code depends on **FMDB** and **FMDB** depends on **SQLite** or **SQLCipher**. `pod lib lint` will not pass if I define a subspec called `common` with our code but without **FMDB**. If I define 2 subspecs: one that imports `FMDB` called `SQLite` and other that imports `FMDB/SQLCipher` called `SQLCipher`, and then I add as a dependency `SQLite` to `common`, `pod lib lint` will succeed but `SQLCipher` is worthless because if someone uses it in his/her `Podfile`: `pod CDTDatastore/SQLCipher`, the only thing he/she will include in his/her workspace will be **FMDB** with **SQLCipher**. Then if I add `common` to `SQLCipher`, `pod lib lint` will succeed but I'll have again the problem I was trying to avoid (I'm importing both **SQLite** and **SQLCipher**).

So:
- Our code can not be compiled in isolation
- It requires `FMDB` or `FMDB/SQLCipher`
- But I do not want to compile the code with both dependencies added at the same time
- Also I want to provide to the user one and only one subspec that, once included in his/her `Podfile`, will add to the workspace all the required source code: **FMDB**, **SQLCipher** & **CDTDatastore**. If he/she types `pod CDTDatastore`, only **FMDB** & **CDTDatastore** will be included.

I can only think of 3 possible solutions:
- Remove `pod lib lint --allow-warnings` from `Rakefile`
- Define a subspec `common` only with our source code. Then another subspec called `SQLite` than depends on `common` and `FMDB` and a third one called `SQLCipher` that depends on `common` and `FMDB/SQLCipher`. Finally I have to replace `pod lib lint --allow-warnings` with:
```
  sh "pod lib lint --allow-warnings --subspec='CDTDatastore/SQLite'"
  sh "pod lib lint --allow-warnings --subspec='CDTDatastore/SQLCipher'"
```
- Do not define `common` and instead copy its content in `SQLite` and `SQLCipher`. It is not necessary to modify `Rakefile`